s4e-backend/src/main/java/pl/cyfronet/s4e/data/repository/SceneRepositoryImpl.java

```diff
@@ -0,0 +1,154 @@
+    private ResultSetHandler<List<Scene>> handler;
+    private QueryRunner runner;
+    private Connection conn;
+
+    public SceneRepositoryImpl(DbProperties dbProperties, TimeHelper timeHelper) throws SQLException {
+        this.dbProperties = dbProperties;
+        this.handler = new BeanListHandler<>(Scene.class, new SceneHandler());
+        this.runner = new QueryRunner();
+        this.conn = DriverManager.getConnection(dbProperties.getUrl(), dbProperties.getUsername(), dbProperties.getPassword());
+    }
+
+    public List<Scene> findAllByParamsMap(Map<String, Object> params) throws SQLException {
+        return runner.query(conn, prepareQuery(params), handler);
+    }
+
+    private String prepareQuery(Map<String, Object> params) {
+        StringBuilder resultQuery = new StringBuilder();
+        resultQuery.append("SELECT s.id, s.product_id FROM Scene s WHERE true");
+        queryBySensingTime(params, resultQuery);
+        queryByIngestionTime(params, resultQuery);
+        queryBySatellitePlatform(params, resultQuery);
+        queryByProductType(params, resultQuery);
+        queryByPolarisation(params, resultQuery);
+        // [string] sensorMode -> ...
+        // [string] collection -> ...
+        queryByCloudCover(params, resultQuery);
+        // [string] timeliness -> ...
+        // [string] instrument -> ...
+        // [string] productLevel -> ...
+        queryByProcessingLevel(params, resultQuery);
+        queryByRelativeOrbitNumber(params, resultQuery);
+        // [NUMBER] absoluteOrbitNumber -> ...
+        // [] inputAuxiliaryFiles -> ...
+        // [] polygon -> field[footprint]
+        return resultQuery.toString();
+    }
+
+    private void queryBySensingTime(Map<String, Object> params, StringBuilder resultQuery) {
+        ZoneParameter tz = new ZoneParameter(ZoneId.of((String) params.getOrDefault("tz", "UTC")));
+        if (params.containsKey("sensingFrom") && params.containsKey("sensingTo")) {
+            // [TIMESTAMP] sensingFrom / sensingTo -> sensing_time
+            resultQuery.append(" AND to_timestamp(s.metadata_content->>'sensing_time', 'YYYY-MM-DD\"T\"HH24:MI:SS\"Z\"')");
+            resultQuery.append("BETWEEN '" + params.get("sensingFrom"));
+            resultQuery.append("' AND '" + params.get("sensingTo"));
+            resultQuery.append("' ");
+        } else {
+            if (params.containsKey("sensingFrom")) {
+                resultQuery.append(" AND to_timestamp(s.metadata_content->>'sensing_time', 'YYYY-MM-DD\"T\"HH24:MI:SS\"Z\"') >= '");
+                resultQuery.append(params.get("sensingFrom"));
+                resultQuery.append("' ");
+            }
+            if (params.containsKey("sensingTo")) {
+                resultQuery.append(" AND to_timestamp(s.metadata_content->>'sensing_time', 'YYYY-MM-DD\"T\"HH24:MI:SS\"Z\"') <= '");
+                resultQuery.append(params.get("sensingTo"));
+                resultQuery.append("' ");
+            }
+        }
```

Comment:
Why you map the metadata names in this way? Shouldn't we map directly, i.e. sensing_time -> sensing_time_from, sensing_time_to, or something like that?

Reply:
resolved: we stay with camelCase on payload side and snake_case in db fields